src/features/todo-form/todo-form.tsx

```diff
@@ -0,0 +1,110 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Todo, TodoSchema } from "../../shared/schemas/todo-schema";
+import { PRIORITIES } from "../../shared/schemas/priority-schema";
+import styles from "./todo-form.module.scss";
+import { FormField } from "./components/form-field/form-field";
+import { useTodoStore } from "../../shared/store/useTodoStore";
+import { capitalize } from "../../shared/util/capitalize";
+import { getTodayDate } from "../../shared/util/getTodayDate";
+import toast from "react-hot-toast";
+import { useNavigate } from "react-router-dom";
+
+export const TodoForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<Todo>({ resolver: zodResolver(TodoSchema) });
+  const { addTodo } = useTodoStore();
+
+  const navigate = useNavigate();
+
+  const onSubmit = (data: Todo) => {
+    addTodo(data);
+    toast.success("Successfully created!");
```

Comment:
Showing a toast notification to the user after executing an action is a nice addition and improves the user experience. Good job!